File: src/CrawlerLogFolder.java
import java.util.Stack;

public class CrawlerLogFolder {
    public static void main(String[] args) {
        String[] logs = {"d1/","d2/","../","d21/","./"};
    }

    // ***************** 1st Method ******************
    // Approach 1: Use stack to store the folder names
    // When encounter "../", pop from stack, when encounter "./", continue
    // Else, push the folder name to stack. Then return stack size
    // Runtime  : 1ms         -> + 99.42%
    // Memory   : 42.30MB     -> + 19.93%
    public int minOperations(String[] logs) {
        Stack<String> stk = new Stack<>();

        for(String s : logs) {
            if(s.equals("../")) {
                if(!stk.isEmpty()) {
                    stk.pop();
                }
            } else if(s.equals("./")) {
                continue;
            } else {
                stk.push(s);
            }
        }
        return stk.size();
    }
    // ***************** End of 1st Method ******************
}

File: CSharp-Solutions/LeetCodeCSharp/LeetCodeCSharp/CrawlerLogFolder.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace LeetCodeCSharp
{
    public class CrawlerLogFolder
    {
        // ***************** 1st Method ******************
        // Approach 1: Use stack and check log[i]
        // if log[i] == "./" -> continue, if log[i] == "../" -> pop, 
        // if not "./" or "../" -> push to stack
        // Before pop check if stack is empty 
        // Runtime  : 58ms      -> + 67.12%
        // Memory   : 40.30MB   -> + 84.93%
        public int MinOperations(string[] logs)
        {
            Stack<string> stk = new Stack<string>();

            for (int i = 0; i < logs.Length; i++)
            {
                if (logs[i].Equals("./")) continue;
                if (logs[i].Equals("../") && stk.Count() > 0) stk.Pop();
                if (!logs[i].Equals("./") && !logs[i].Equals("../")) stk.Push(logs[i]);
            }

            return stk.Count();
        }
        // ***************** End of 1st Method ******************
    }
}